extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,972 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
 
-1. [State](#state): A document-scope, mutable JSON state. In the example above, the state is empty before tapping the button. After tapping the button, the state is `{foo: 'amp-bind'}`.
-2. [Expressions](#expressions): These are JavaScript-like expressions that can reference the **state**. The example above has a single expression, `'Hello ' + foo`, which concatenates the string literal `'Hello '` and the state variable `foo`.
-   There is a limit of 100 operands what can be used in an expression.
-3. [Bindings](#bindings): These are special attributes of the form `[property]` that link an element's property to an **expression**. The example above has a single binding, `[text]`, which updates the `<p>` element's text every time the expression's value changes.
+When the user taps/clicks the button:
 
-`amp-bind` takes special care to ensure speed, security and performance on AMP pages.
+1. It triggers the `tap` event.
+1. The `tap` event invokes the `AMP.setState()` method.
+1. The `AMP.setState()` methods sets the `foo` **state variable** to the value of `Interactivity`.
+1. The state is no longer empty, so the page updates the bound property to its state.
+
+[tip type="note"]
+Calling `AMP.setState()` in some examples may set or change states of other examples on page. Refresh this page to see examples before `AMP.setState()`.
+[/tip]
 
 ### A slightly more complex example
 
 [filter formats="websites, stories, ads"]
 
+[example preview="top-frame" playground="true" imports="amp-bind"]
+
+```html
+<head>
+  <style amp-custom>
+    .greenBorder {
+      border: 5px solid green;
+    }
+    .redBorder {
+      border: 5px solid red;
+    }
+  </style>
+</head>
+<body>
+  <amp-state id="theFood">
+    <script type="application/json">
+      {
+        "cupcakes": {
+          "imageUrl": "https://amp.dev/static/samples/img/image2.jpg",
+          "style": "greenBorder"
+        },
+        "sushi": {
+          "imageUrl": "https://amp.dev/static/samples/img/image3.jpg",
+          "style": "redBorder"
+        }
+      }
+    </script>
+  </amp-state>
+  <div class="greenBorder" [class]="theFood[currentMeal].style">
+    <p>Each food has a different border color.</p>
+    <p [text]="'I want to eat ' + currentMeal + '.'">I want to eat cupcakes.</p>
+    <amp-img
+      width="300"
+      height="200"
+      src="https://amp.dev/static/samples/img/image2.jpg"
+      [src]="theFood[currentMeal].imageUrl"
+    >
+    </amp-img>
+    <button on="tap:AMP.setState({currentMeal: 'sushi'})">Set to sushi</button>
+    <button on="tap:AMP.setState({currentMeal: 'cupcakes'})">
+      Set to cupcakes
+    </button>
+  </div>
+</body>
+```
+
+[/example]
+
+In the example above:
+
+- The `<amp-state>` component declares state using a JSON string literal. It has an `id` of `theFood` to allow us to reference the defined data. But because `<amp-bind>` does not evaluate `<amp-state>` on page load, the **state** is empty.
```

Comment:
Can likely drop 'string literal' or 'JSON' here.

```suggestion
- The `<amp-state>` component declares state using JSON. The `<amp-state>` element has an `id` of `theFood` to allow us to reference the defined data. But because `<amp-bind>` does not evaluate `<amp-state>` on page load, the **state** is empty.
```